### DISCUSSION TITLE
Mantine 7.10.0 compatibility with React 19

### DISCUSSION THREAD

**Discussion by iam-sachinverma** [mantinedev/mantine#6323] [Q&A]
**_React 19 supports ref as a prop, leading to the deprecation of element.ref in favor of element.props.ref_**
I recently upgraded to React 19 and encountered a warning with the Mantine Provider

![image](https://github.com/mantinedev/mantine/assets/83693262/0f1676ee-2dac-4884-8438-ee485e359c6b)
![image](https://github.com/mantinedev/mantine/assets/83693262/ff5c94aa-8732-4613-aa48-2dc616c6a6d7)

Can anyone provide guidance on how to resolve this issue?

**Comment by iam-sachinverma** [comment]
Same issue in react@18.3 also

**Comment by BrendanC23** [comment]
Bump.

**Comment by MariuzM** [comment]
same with ^7.11.0

**Comment by gardner** [comment]
The recently released v9 requires React 19:

> Mantine 9.x requires React 19.2 or later. If your project uses an older React version, you need to update it before migrating to Mantine 9.x. If you cannot update React to 19.2+ yet, you can continue using Mantine 8.x until you are ready to update React and migrate to Mantine 9.x.

From [guides/8x-to-9x.mdx](https://github.com/mantinedev/mantine/blob/7e9f9435ddabc67bcfd813cea24ea254ae82db4d/apps/mantine.dev/src/pages/guides/8x-to-9x.mdx#prerequisites)
